compiler/rustc_typeck/src/check/op.rs

```diff
@@ -959,6 +959,13 @@ fn suggest_impl_missing(err: &mut DiagnosticBuilder<'_>, ty: Ty<'_>, missing_tra
                 "an implementation of `{}` might be missing for `{}`",
                 missing_trait, ty
             ));
+            // This checks if the missing trait is PartialEq to suggest adding a derive
+            if missing_trait.ends_with("PartialEq") {
```

Comment:
Consider changing the check to only look for `PartialEq` and nothing else. `MyPartialEq` and `FooPartialEq` will trigger this.

Reply:
There's no real risk of this matching `MyPartialEq` or `FooPartialEq` here, as far as I can gather the `missing_trait` argument always comes from a string literal in the calling function.

However, I'd prefer we didn't rely on string comparisons here - I think it should be possible to modify the calling function to pass something more precise to this function - I'd probably try using `get_diagnostic_item` and providing the `DefId` that it returns to this function (that might not work if diagnostic items aren't available in this part of the compiler, I've forgotten exactly - but you could always just use `LangItem` or `Symbol` instead).

Alternatively, you could add a `can_derive` argument and thread that value through, to make this conditional more broadly applicable than just `PartialEq` (or make both of my suggestions here).

Reply:
This function is only used with binop traits so we won't have any conflicts. Anyways I fixed the issue replacing the `ends_with` with a strict check.